src/views/inspector/InspectorViewProvider.ts

```diff
@@ -27,27 +28,40 @@ export class InspectorViewProvider implements vscode.WebviewViewProvider {
     if (!folder) {
       return
     }
-    this.variablesCLIController = new VariablesCLIController(folder)
-    this.featuresCLIController = new FeaturesCLIController(folder)
 
-    this.variablesCLIController.getAllVariables().then((variables) => {
-      this.variables = variables
-      if (Object.values(this.variables).length) {
-        this.selectedKey = Object.values(this.variables)[0].key
-      }
-    })
+    this.variables = StateManager.getFolderState(folder.name, KEYS.VARIABLES) || {}
+    this.features = StateManager.getFolderState(folder.name, KEYS.FEATURES) || {}
+    this.matches = StateManager.getFolderState(folder.name, KEYS.CODE_USAGE_KEYS) || {}
+    
+    const variablesCLIController = new VariablesCLIController(folder)
+    const featuresCLIController = new FeaturesCLIController(folder)
 
-    this.featuresCLIController.getAllFeatures().then((features) => {
-      this.features = features
-    })
+    if (!Object.keys(this.variables).length) {
+      variablesCLIController.getAllVariables().then((variables) => {
```

Comment:
the variables controller already does this check for a value in the state manager before calling the cli

Reply:
so then the idea is for the StateManager to deal with fetching variables / features at startup of the extension? because I added this check because the features wasn't getting fetched yet at this point

Reply:
The variable and features cli controllers are setup so that you can always call `getAllVariables`/`getAllFeatures` and it will read from the StateManager if the value is available, otherwise it will call the CLI.

The usages controller has not been setup like this, but for consistency maybe it should be.